staging/src/k8s.io/kubectl/pkg/cmd/util/helpers.go

```diff
@@ -396,6 +396,17 @@ func AddValidateFlags(cmd *cobra.Command) {
 	cmd.Flags().Bool("validate", true, "If true, use a schema to validate the input before sending it")
 }
 
+// AddUserFlags add "user" flag to local flags, then the user can see the usage of "user" flag in the help message
+func AddUserFlags(cmd *cobra.Command, usage string) {
+	localFlags := cmd.LocalFlags()
+	localFlags.StringArray("user", []string{}, usage)
+	// now the root command has defined persistent flags "user".
+	// you can see it through "kubectl option". the create clusterrole and rolebindding command cannot
+	// use the inherited "user" flag in order to maintain compatibility.
+	// so add "user" flag to Flags() again to overwrite inherited flags.
+	cmd.Flags().AddFlagSet(localFlags)
```

Comment:
this is pretty twisty... I wouldn't have expected this to be necessary... what is omitting the local flag from the usage text? do we need to fix something about our help template?

Reply:
when you use `cmd.Flags().StringArray("user", []string{}, "Usernames to bind to the clusterrole")` , the `--user Usernames to bind to the role` flag not in localFlags but in Flags(), because InheritedFlags() has `--user The name of the kubeconfig user to use`.
Flags() contains all flags, some of which are undesirable for subcommands. modifying the template does not work, only let `--user Usernames to bind to the role` appears in LocalFlags().

Follow code also can work
``
+ cmd.LocalFlags().StringArray("user", []string{}, "Usernames to bind to the clusterrole")
cmd.Flags().StringArray("user", []string{}, "Usernames to bind to the clusterrole")
```